backend/src/sdk.cpp

```diff
@@ -209,6 +240,147 @@ restinio::request_handling_status_t SDK::handleTxSDKCall(const restinio::request
+            mClient->SetTx(frequency, !mClient->GetTxState(frequency));
+        } else {
+            auto tx = json["value"]["tx"].get<bool>();
+
+            TRACK_LOG_INFO("Setting tx for {} to {}", frequency, tx);
+            mClient->SetTx(frequency, tx);
+        }
+    }
+
+    if (json["value"].contains("xc") && UserSession::isATC) {
+        // xc is either a "toggle" string or a boolean value
+        if (json["value"]["xc"].is_string() && json["value"]["xc"] == "toggle") {
+            TRACK_LOG_INFO("Toggling xc for {}", frequency);
+            mClient->SetXc(frequency, !mClient->GetXcState(frequency));
+        } else {
+            auto xc = json["value"]["xc"].get<bool>();
+
+            TRACK_LOG_INFO("Setting xc for {} to {}", frequency, xc);
+            mClient->SetXc(frequency, xc);
+        }
+    }
+
+    if (json["value"].contains("xca") && UserSession::isATC) {
+        // xca is either a "toggle" string or a boolean value
+        if (json["value"]["xca"].is_string() && json["value"]["xca"] == "toggle") {
+            TRACK_LOG_INFO("Toggling xca for {}", frequency);
+            mClient->SetCrossCoupleAcross(
+                frequency, !mClient->GetCrossCoupleAcrossState(frequency));
+        } else {
+            auto xca = json["value"]["xca"].get<bool>();
+
+            TRACK_LOG_INFO("Setting xca for {} to {}", frequency, xca);
+            mClient->SetCrossCoupleAcross(frequency, xca);
+        }
+    }
+
+    if (json["value"].contains("headset")) {
+        // headset is either a "toggle" string or a boolean value
+        if (json["value"]["headset"].is_string() && json["value"]["headset"] == "toggle") {
+            TRACK_LOG_INFO("Toggling headset for {}", frequency);
+            mClient->SetOnHeadset(frequency, !mClient->GetOnHeadset(frequency));
+        } else {
+            auto headset = json["value"]["headset"].get<bool>();
+
+            TRACK_LOG_INFO("Setting headset for {} to {}", frequency, headset);
+            mClient->SetOnHeadset(frequency, headset);
+        }
+    }
+
+    // Send updated info to connected clients
+    auto stateJson = this->buildStationStateJson(std::nullopt, frequency);
+    this->publishStationState(stateJson);
+    NapiHelpers::callElectron("station-state-update", stateJson.dump());
+}
+
+void SDK::handleGetStationStates()
+{
+    std::vector<nlohmann::json> stationStates;
+
+    // Collect the states for all the radios
+    auto allRadios = mClient->getRadioState();
+    for (const auto& [frequency, state] : allRadios) {
+        stationStates.push_back(this->buildStationStateJson(state.stationName, frequency));
+    }
+
+    // Send the message
+    nlohmann::json jsonMessage
+        = WebsocketMessage::buildMessage(WebsocketMessageType::kStationStates);
+    jsonMessage["value"]["stations"] = stationStates;
+    this->broadcastOnWebsocket(jsonMessage.dump());
+}
+
+void SDK::handleGetStationState(const std::string& callsign)
```

Comment:
This gets called by the streamdeck plugin to get the state for a single station when a new button is added.